src/Integration.Vsix/SonarLintDaemon.cs

```diff
@@ -0,0 +1,140 @@
+        {
+        }
+
+        public SonarLintDaemon(string version, string storagePath, string tmpPath)
+        { 
+            this.version = version;
+            this.tmpPath = tmpPath;
+            this.storagePath = storagePath;
+        }
+
+        public void Install()
+        {
+            Download();
+            Unzip();
+        }
+
+        public bool IsRunning()
+        {
+            if (process == null)
+            {
+                return false;
+            }
+            return !process.HasExited;
+        }
+
+        public void Start()
+        {
+            if (IsRunning())
+            {
+                throw new InvalidOperationException("Process already running");
+            }
+            if (!IsInstalled())
+            {
+                throw new InvalidOperationException("Daemon is not installed");
+            }
+
+            if (process != null)
+            {
+                process.Close();
+            }
+
+            process = new Process();
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.FileName = GetExePath();
+            process.StartInfo.Arguments = GetCmdArgs();
+            process.StartInfo.CreateNoWindow = true;
+            process.Start();
+        }
+
+        public void Stop()
+        {
+            if (!IsRunning())
+            {
+                return;
+                // throw exception?
+            }
+            process.Kill();
+            process.WaitForExit();
+        }
+
+        public bool IsInstalled()
+        {
+            return Directory.Exists(GetInstallationPath()) && File.Exists(GetExePath());
+        }
+
+        #region private methods
+
+        private void Download()
+        {
+            string uri = string.Format(uriFormat, version);
+            using (var client = new WebClient())
+            {
+                client.DownloadFile(uri, GetZipFilePath());
+            }
+        }
+
+        private void Unzip()
+        {
+            if (Directory.Exists(GetInstallationPath()))
+            {
+                Directory.Delete(GetInstallationPath(), true);
+            }
+            ZipFile.ExtractToDirectory(GetZipFilePath(), storagePath);
+        }
+
+        private string GetInstallationPath()
+        {
+            return Path.Combine(storagePath, "sonarlint-daemon-" + version + "-windows");
+        }
+
+        private string GetZipFilePath()
+        {
+            return Path.Combine(tmpPath, "sonarlint-daemon-" + version + "-windows.zip");
+        }
+
+        private string GetExePath()
+        {
+            return Path.Combine(GetInstallationPath(), "jre", "bin", "java.exe");
+        }
+
+        private string GetCmdArgs()
+        {
+            string jarPath = Path.Combine(GetInstallationPath(), "lib", string.Format("sonarlint-daemon-{0}.jar", version));
```

Comment:
This 

    string.Format("sonarlint-daemon-{0}.jar", version) 

could also be:

    $"sonarlint-daemon-{version}.jar"